src/Microsoft.TemplateEngine.Utils/DefaultTemplateEngineHost.cs

```diff
@@ -62,30 +72,71 @@ public void UpdateLocale(string newLocale)
 
         public virtual IReadOnlyList<KeyValuePair<Guid, Func<Type>>> BuiltInComponents => _hostBuiltInComponents;
 
+        public ILogger Logger => _logger;
+
         public virtual void LogMessage(string message)
         {
-            //Console.WriteLine("LogMessage: {0}", message);
-            Console.WriteLine(message);
+            Logger.LogInformation(message);
         }
 
         public virtual void OnCriticalError(string code, string message, string currentFile, long currentPosition)
         {
+            StringBuilder logMessage = new StringBuilder();
+            logMessage.Append(message);
+            if (!string.IsNullOrWhiteSpace(code))
+            {
+                logMessage.Append($" Error code:{code}.");
+            }
+            if (!string.IsNullOrWhiteSpace(currentFile))
+            {
+                logMessage.Append($" File:{currentFile}, position: {currentPosition}.");
+            }
+            Logger.LogError(logMessage.ToString());
         }
 
         public virtual bool OnNonCriticalError(string code, string message, string currentFile, long currentPosition)
         {
-            LogMessage(string.Format($"Error: {message}"));
+            StringBuilder logMessage = new StringBuilder();
+            logMessage.Append(message);
+            if (!string.IsNullOrWhiteSpace(code))
+            {
+                logMessage.Append($" Error code:{code}.");
+            }
+            if (!string.IsNullOrWhiteSpace(currentFile))
+            {
+                logMessage.Append($" File:{currentFile}, position: {currentPosition}.");
+            }
+            Logger.LogWarning(logMessage.ToString());
             return false;
         }
 
         public virtual bool OnParameterError(ITemplateParameter parameter, string receivedValue, string message, out string newValue)
         {
+            StringBuilder logMessage = new StringBuilder();
```

Comment:
Should any of this be localized?

Reply:
I'm not sure if we should localize any logs tbh